drivers/modem/gsm_ppp.c

```diff
@@ -474,6 +611,32 @@ static void gsm_finalize_connection(struct gsm_modem *gsm)
 		}
 		modem_cmd_handler_tx_unlock(&gsm->context.cmd_handler);
 	}
+
+#if defined(CONFIG_GSM_MUX)
+		k_work_schedule(&rssi_work_handle, K_NO_WAIT);
```

Comment:
This should only be called if gsm muxing actually is enabled. To ensure this, I suggest to call it right after modem_cmd_handler_tx_unlock().

Reply:
addressed now